ci-operator/step-registry/cucushift/installer/check/aws/private/cucushift-installer-check-aws-private-commands.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+REGION="${LEASED_RESOURCE}"
+INFRA_ID=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+ret=0
+
+#Check there's no external load balancer created in this cluster
+
+EXT_LB=$(aws --region ${REGION} elbv2 describe-load-balancers |jq --arg var "${INFRA_ID}" -r '.LoadBalancers[] | select(.LoadBalancerName == "\($var)-ext")')
+
+if [[ -z ${EXT_LB} ]]; then
+   echo "PASS: no external LB detected in this cluster"
+else
+   echo "Error: external LB detected in this cluster"
+   ret=$((ret + 1))
+fi
+
+#No public Route 53 DNS records that matches the baseDomain for the cluster
+PUBLIC_ZONE_ID=$(aws route53 list-hosted-zones-by-name | jq --arg name "${BASE_DOMAIN}." -r '.HostedZones | .[] | select(.Name=="\($name)") | .Id' | awk -F / '{printf $3}')
+API_RECORD_SET=$(aws route53 list-resource-record-sets --hosted-zone-id=${PUBLIC_ZONE_ID} --output json | jq --arg dns "api.${CLUSTER_NAME}.${BASE_DOMAIN}." '.ResourceRecordSets[] | select(.Name == "\($dns)")')
+INGRESS_RECORD_SET=$(aws route53 list-resource-record-sets --hosted-zone-id=${PUBLIC_ZONE_ID} --output json | jq --arg dns "\\052.apps.${CLUSTER_NAME}.${BASE_DOMAIN}." '.ResourceRecordSets[] | select(.Name == "\($dns)")')
```

Comment:
I am not sure if we can use `${CLUSTER_NAME}.${BASE_DOMAIN}` to filter the records, if yes, that would be more accurate to avoid some unexpected public dns records are created.

Reply:
yes, we can use grep to filter ${CLUSTER_NAME}.${BASE_DOMAIN}  directly, without using "jq's select" to run the matching.